### ISSUE TITLE
malformed data coming out of data processing 🫠

### ISSUE THREAD

**Issue by biblicabeebli** [onnela-lab/beiwe-backend#373]
There has been some churn in the data processing code over the last week, it looks like I introduced a bug.

looks like the code that inserts the human-readable timestamp is triggering whenever a file gets reprocessed, causing data to look like this if it is reprocessed many times (this is accelerometer data I have personally generated).
```
'1711505884462,2024-03-27T02:18:04.462,unknown,-0.02856 4453125,-0.008636474609375,-1.023468017578125'
'1711505884462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,unknown,-0.028564453125,-0.008636474609375,-1.023468017578125'
'1711505884462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,unknown,-0.028564453125,-0.008636474609375,-1.023468017578125'
'1711505884462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,unknown,-0.028564453125,-0.008636474609375,-1.023468017578125'
'1711505884462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,unknown,-0.028564453125,-0.008636474609375,-1.023468017578125'
'1711505884462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,unknown,-0.028564453125,-0.008636474609375,-1.023468017578125'
'1711505884462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,unknown,-0.028564453125,-0.008636474609375,-1.023468017578125'
'1711505884462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,2024-03-27T02:18:04.462,unknown,-0.028564453125,-0.008636474609375,-1.023468017578125'

```

**Comment by biblicabeebli** [comment]
I have stopped data processing on all our servers until this is rectified, recommend any other admins seeing this do the same if you have updated in the past two weeks.

**Comment by biblicabeebli** [comment]
The issue has been resolved.  As part of a refactor last week a function that does some critical work unpacking csv files so that we can merge data into hourly chunks was updated without the realization that this function was used to pull in both already processed data (data than had already been "cleaned") and unprocessed data.  As a result one of the operations, inserting a human readable timestamp, got duplicated any time existing data got updated in a new processing run.

As a result of reviewing the data processing code and implementing a fix for the malformed data I was able to fix a memory leak that has existed for at least 8 years.  File processing is now substantially more efficient.

There is a bunch of code mess that this has created that will need to be cleaned up, but a bunch of older mess was cleaned up to make room for the new mess, so it's a bit of a wash.  Some of the code complexity has been reduced due to the _removal_ of multithreading. This decision was made because it further reduces memory overhead.

As yet another big change that I threw in is we now save app version history (Just before the 2.5 ios release got pushed) and associate that with file uploads.  We can use this to transition fixes that have occurred on the backend to the apps and then eventually remove the extra complexity from the backend altogether.
